infrastructure/templates/public-api/main.bicep

```diff
@@ -103,51 +100,72 @@ var commonResourcePrefix = '${subscription}-ees'
 // The resource prefix for resources created in the ARM template.
 var legacyResourcePrefix = subscription
 
-
-//
-// Define full names for the pre-existing resources managed by the ARM template.
-//
-var adminAppName = '${legacyResourcePrefix}-as-ees-admin'
-var publisherAppFullName = '${legacyResourcePrefix}-fa-ees-publisher'
-var keyVaultName = '${legacyResourcePrefix}-kv-ees-01'
-var vNetName = '${legacyResourcePrefix}-vnet-ees'
-var alertsGroupName = '${legacyResourcePrefix}-ag-ees-alertedusers'
-var acrName = 'eesacr'
+var resourceNames = {
+  existingResources: {
+    adminApp: '${legacyResourcePrefix}-as-ees-admin'
+    publisherFunction: '${legacyResourcePrefix}-fa-ees-publisher'
+    keyVault: '${legacyResourcePrefix}-kv-ees-01'
+    vNet: '${legacyResourcePrefix}-vnet-ees'
+    alertsGroup: '${legacyResourcePrefix}-ag-ees-alertedusers'
+    acr: 'eesacr'
+    coreStorageAccount: '${legacyResourcePrefix}saeescore'
+    subnets: {
+      adminApp: '${legacyResourcePrefix}-snet-as-ees-admin'
+      appGateway: '${commonResourcePrefix}-snet-${abbreviations.networkApplicationGateways}-01'
+      containerAppEnvironment: '${commonResourcePrefix}-snet-${abbreviations.appManagedEnvironments}-01'
+      dataProcessor: '${publicApiResourcePrefix}-snet-${abbreviations.webSitesFunctions}-processor'
+      dataProcessorPrivateEndpoints: '${publicApiResourcePrefix}-snet-${abbreviations.webSitesFunctions}-processor-pep'
+      publisherFunction: '${legacyResourcePrefix}-snet-a-ees-publisher'
+      psqlFlexibleServer: '${commonResourcePrefix}-snet-${abbreviations.dBforPostgreSQLServers}'
+    }
+  }
+  sharedResources: {
+    appGateway: '${commonResourcePrefix}-${abbreviations.networkApplicationGateways}-01'
+    appGatewayIdentity: '${commonResourcePrefix}-${abbreviations.managedIdentityUserAssignedIdentities}-${abbreviations.networkApplicationGateways}-01'
+    containerAppEnvironment: '${commonResourcePrefix}-${abbreviations.appManagedEnvironments}-01'
+    logAnalyticsWorkspace: '${commonResourcePrefix}-${abbreviations.operationalInsightsWorkspaces}'
+    postgreSqlFlexibleServer: '${commonResourcePrefix}-${abbreviations.dBforPostgreSQLServers}'
+  }
+  publicApi: {
+    apiApp: '${publicApiResourcePrefix}-${abbreviations.appManagedEnvironments}-api'
+    apiAppIdentity: '${publicApiResourcePrefix}-${abbreviations.managedIdentityUserAssignedIdentities}-${abbreviations.appManagedEnvironments}-api'
+    appInsights: '${publicApiResourcePrefix}-${abbreviations.insightsComponents}'
+    dataProcessor: '${publicApiResourcePrefix}-${abbreviations.webSitesFunctions}-processor'
+    dataProcessorIdentity: '${publicApiResourcePrefix}-${abbreviations.managedIdentityUserAssignedIdentities}-${abbreviations.webSitesFunctions}-processor'
+    // TODO - would this better be '${publicApiResourcePrefix}-${abbreviations.webServerFarms}-${abbreviations.webSitesFunctions}-processor' 
```

Comment:
Something still to decide on here before this change goes in?

Reply:
I don't think so - I think this is to go in with all of the potential naming convention changes we *could* do but don't have to :)